### PR TITLE
Additional information for documentation on macros

### PR DESCRIPTION
**Pull request by gsps1**
JIRA : https://issues.cask.co/browse/HYDRATOR-300

### CODE REVIEW

**Comment by awholegunch** [comment]
Running a Quick Build: http://builds.cask.co/browse/CDAP-DQB131-2

**Build failed, but unrelated to this PR.**

Pages of interest (when build has completed): 
- **Creating Pipelines:** http://builds.cask.co/artifact/CDAP-DQB131/shared/build-2/Docs-HTML/4.0.0-SNAPSHOT/en/hydrator-manual/creating-pipelines.html#field-macro-substitution

File: cdap-docs/hydrator-manual/source/developing-plugins/creating-a-plugin.rst [hunk 1]
```diff
@@ -63,12 +63,13 @@ the types, only one method is required to be implemented::
...
+- ``prepareRun()``: Used to configure the input for each run of the pipeline.
+  If the fieldName for a stream or dataset is a macro, their creation will happen during this stage. 
+  This is called by the client that will submit the job for the pipeline run.
 - ``onRunFinish()``: Used to run any required logic at the end of a pipeline run. This is called
   by the client that submitted the job for the pipeline run.
-- ``configurePipeline()``: Used to create any streams or datasets or perform any validation
-  on the application configuration that are required by this plugin.
+- ``configurePipeline()``: Used to perform any validation on the application configuration that is required
+  by this plugin or create any streams or datasets if the fieldName for a stream or dataset is not a macro.
 - ``initialize()``: Initialize the Batch Source. Guaranteed to be executed before any call
```
Comment (gsps1) [inline cdap-docs/hydrator-manual/source/developing-plugins/creating-a-plugin.rst:76]: these methods are duplicated for multiple type of plugins, i just updated it for batch-source, though its applicable for all the types.

Comment (awholegunch) [reply cdap-docs/hydrator-manual/source/developing-plugins/creating-a-plugin.rst:76]: Added to all but the deprecated instances.

**Comment by awholegunch** [comment]
Running a Quick Build: http://builds.cask.co/browse/CDAP-DQB131-3

**Build failed, but unrelated to this PR.**

Pages of interest (when build has completed): 
- **Creating Pipelines:** http://builds.cask.co/artifact/CDAP-DQB131/shared/build-3/Docs-HTML/4.0.0-SNAPSHOT/en/hydrator-manual/creating-pipelines.html#field-macro-substitution

**Comment by gsps1** [comment]
did the build pass ?

**Comment by awholegunch** [comment]
No; "Build failed, but unrelated to this PR.".

**Comment by gsps1** [comment]
LGTM

**Pull request merged**